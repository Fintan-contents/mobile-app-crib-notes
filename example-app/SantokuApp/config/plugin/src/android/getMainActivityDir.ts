import {AndroidConfig} from '@expo/config-plugins';
import path from 'path';

export async function getMainActivityDir(projectRoot: string) {
  const mainActivity = await AndroidConfig.Paths.getMainActivityAsync(projectRoot);
  return path.dirname(mainActivity.path);
}
