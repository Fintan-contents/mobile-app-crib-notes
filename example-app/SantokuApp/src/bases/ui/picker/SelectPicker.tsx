import React from 'react';
import {Pressable, StyleSheet, TextInput, TextInputProps as RNETextInputProps, View, ViewProps} from 'react-native';

import {DefaultPickerAccessory, DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdrop, PickerBackdropProps} from './PickerBackdrop';
import {PickerContainer, PickerContainerProps} from './PickerContainer';
import {SelectPickerItems, SelectPickerItemsProps} from './SelectPickerItems';
import {useSelectPicker} from './useSelectPicker';

export type Item<T> = {
  label: string;
  value: T;
  inputLabel?: string;
  key?: React.Key;
  color?: string;
  fontFamily?: string;
};

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;
export type SelectPickerProps<ItemT> = {
  /**
   * SelectPickerに表示するアイテム
   */
  items: Item<ItemT>[];
  /**
   * 選択されたアイテムのKey
   */
  selectedItemKey?: React.Key | ItemT;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (selectedItem?: Item<ItemT>) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   */
  onDismiss?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDelete?: (selectedItem?: Item<ItemT>) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onCancel?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDone?: (selectedItem?: Item<ItemT>) => void;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、{@link DefaultPickerAccessory}がデフォルトで表示されます。
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、{@link SelectPickerItems}がデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * Keyを作成する関数
   */
  keyExtractor?: ((item: Item<ItemT>, index: number) => string) | undefined;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   */
  pickerProps?: Omit<SelectPickerItemsProps<ItemT>, 'items' | 'selectedValue' | 'onValueChange' | 'keyExtractor'>;
  /**
   * PickerBackdropのProps
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

const DEFAULT_DURATION = 300;

export const SelectPicker = <ItemT extends unknown>(props: SelectPickerProps<ItemT>) => {
  const {isVisible, inputValue, handleBackdropPress, onValueChange, open, handleDelete, handleCancel, handleDone} =
    useSelectPicker<ItemT>(props);
  const {
    items,
    selectedItemKey,
    placeholder,
    textInputProps,
    keyExtractor,
    pickerAccessoryComponent,
    pickerItemsComponent,
    textInputComponent,
    pickerItemsContainerProps: {style: pickerItemsContainerStyle, ...pickerItemsContainerProps} = {},
    pickerProps,
    pickerBackdropProps: {
      fadeInDuration = DEFAULT_DURATION,
      fadeOutDuration = DEFAULT_DURATION,
      ...pickerBackdropProps
    } = {},
    pickerContainerProps: {
      style: pickerContainerStyle,
      slideInDuration = DEFAULT_DURATION,
      slideOutDuration = DEFAULT_DURATION,
      ...pickerContainerProps
    } = {},
    pickerAccessoryProps,
  } = props;

  return (
    <>
      <PickerBackdrop
        isVisible={isVisible}
        onPress={handleBackdropPress}
        fadeInDuration={fadeInDuration}
        fadeOutDuration={fadeOutDuration}
        {...pickerBackdropProps}>
        <PickerContainer
          isVisible={isVisible}
          style={[styles.pickerContainer, pickerContainerStyle]}
          slideInDuration={slideInDuration}
          slideOutDuration={slideOutDuration}
          {...pickerContainerProps}>
          {pickerAccessoryComponent ? (
            pickerAccessoryComponent
          ) : (
            <DefaultPickerAccessory
              onDelete={handleDelete}
              onCancel={handleCancel}
              onDone={handleDone}
              {...pickerAccessoryProps}
            />
          )}
          <View {...pickerItemsContainerProps}>
            {pickerItemsComponent ? (
              pickerItemsComponent
            ) : (
              <SelectPickerItems
                selectedValue={selectedItemKey}
                items={items}
                onValueChange={onValueChange}
                keyExtractor={keyExtractor}
                {...pickerProps}
              />
            )}
          </View>
        </PickerContainer>
      </PickerBackdrop>
      <Pressable onPress={open} testID="pressableContainer">
        <View pointerEvents="box-only">
          {textInputComponent ? (
            textInputComponent
          ) : (
            // テキスト入力とスタイルを合わせるために、TextではなくTextInputを使用する
            <TextInput placeholder={placeholder} value={inputValue} editable={false} {...textInputProps} />
          )}
        </View>
      </Pressable>
    </>
  );
};

const styles = StyleSheet.create({
  pickerContainer: {
    backgroundColor: 'white',
  },
});
