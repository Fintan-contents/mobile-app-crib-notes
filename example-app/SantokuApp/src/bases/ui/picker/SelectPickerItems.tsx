import {PickerProps} from '@react-native-picker/picker';
import {log} from 'bases/logging';
import React from 'react';
import {Platform, StyleProp, TextStyle} from 'react-native';

import {Item} from './SelectPicker';

export type SelectPickerItemsProps<ItemT> = Omit<
  PickerProps<React.Key | ItemT>,
  'enabled' | 'mode' | 'prompt' | 'dropdownIconColor' | 'dropdownIconRippleColor' | 'onFocus' | 'onBlur' | 'style'
> & {
  items: Item<ItemT>[];
  keyExtractor?: ((item: Item<ItemT>, index: number) => string) | undefined;
  /**
   * Pickerのスタイル
   * @platform ios
   */
  style?: StyleProp<TextStyle>;
  /**
   * アイテムの高さ
   * @platform android
   */
  itemHeight?: number;
  /**
   * 選択状態のアイテムのテキストカラー
   * @platform android
   */
  activeColor?: string;
  /**
   * 未選択状態のアイテムのテキストカラー
   * @platform android
   */
  inactiveColor?: string;
  /**
   * FlatListのtestID
   * @platform android
   */
  flatListTestID?: string;
  /**
   * SeparatorのtestID
   * @platform android
   */
  separatorTestID?: string;
  /**
   * SelectPickerItemのPressableのtestIDプレフィックス
   * @platform android
   */
  itemPressableTestIDPrefix?: string;
  /**
   * SelectPickerItemのTextのtestIDプレフィックス
   * @platform android
   */
  itemTextTestIDPrefix?: string;
};

export const SelectPickerItems = <ItemT extends unknown>(props: SelectPickerItemsProps<ItemT>) => {
  React.useEffect(() => {
    log.warn(`SelectPickerItems is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
