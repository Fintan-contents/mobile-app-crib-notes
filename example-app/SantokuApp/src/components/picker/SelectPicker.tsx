import {PickerProps} from '@react-native-picker/picker';
import {log} from 'framework/logging';
import React from 'react';
import {Platform, StyleProp, TextStyle, TextInputProps as RNETextInputProps, ViewProps} from 'react-native';

import {DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdropProps} from './PickerBackdrop';
import {PickerContainerProps} from './PickerContainer';

export type Item<T> = {
  label: string;
  value: T;
  inputLabel?: string;
  key?: React.Key;
  color?: string;
  fontFamily?: string;
  /**
   * アイテムに指定するスタイル
   * 以下のプロパティのみ使用できます。
   * - 'color'
   * - 'backgroundColor'
   * - 'fontSize'
   * - 'fontFamily'
   *
   * @platform android
   */
  style?: StyleProp<Pick<TextStyle, 'color' | 'backgroundColor' | 'fontSize' | 'fontFamily'>>;
};

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;
export type SelectPickerProps<ItemT> = {
  /**
   * SelectPickerに表示するアイテム
   */
  items: Item<ItemT>[];
  /**
   * 選択されたアイテムのKey
   */
  selectedItemKey?: React.Key;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (selectedItem?: Item<ItemT>) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   *
   * @platform ios
   */
  onDismiss?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * @platform ios
   */
  onDelete?: (selectedItem?: Item<ItemT>) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * @platform ios
   */
  onCancel?: (selectedItem?: Item<ItemT>) => void;
  /**
   * [iOS]
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * [Android]
   * アイテムが選択された場合に呼び出される関数
   */
  onDone?: (selectedItem?: Item<ItemT>) => void;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、PickerAccessoryがデフォルトで表示されます。
   *
   * @platform ios
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、SelectPickerItemsがデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * Keyを作成する関数
   */
  keyExtractor?: ((item: ItemT, index: number) => React.Key) | undefined;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   *
   * @platform ios
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   */
  pickerProps?: Omit<PickerProps<ItemT>, 'selectedValue' | 'onValueChange'>;
  /**
   * PickerBackdropのProps
   *
   * @platform ios
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   *
   * @platform ios
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * @platform ios
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

export const SelectPicker = <ItemT extends unknown>(props: SelectPickerProps<ItemT>) => {
  React.useEffect(() => {
    log.warn(`SelectPicker is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
