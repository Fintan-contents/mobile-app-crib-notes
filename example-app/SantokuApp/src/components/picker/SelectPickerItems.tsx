import {PickerProps} from '@react-native-picker/picker';
import {log} from 'framework/logging';
import React from 'react';
import {Platform} from 'react-native';

import {Item} from './SelectPicker';

export type SelectPickerItemsProps = PickerProps<React.Key> & {
  items: Item<React.Key>[];
  keyExtractor?: ((item: React.Key, index: number) => React.Key) | undefined;

  // android
  initialValue?: React.Key;
  itemHeight?: number;
  preferredNumVisibleRows?: number;
};

export const SelectPickerItems = (props: SelectPickerItemsProps) => {
  React.useEffect(() => {
    log.warn(`SelectPickerItems is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
