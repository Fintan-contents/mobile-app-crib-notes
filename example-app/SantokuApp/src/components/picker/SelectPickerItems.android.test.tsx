import {fireEvent, render, screen} from '@testing-library/react-native';
import React from 'react';
import {FlatListProps, PressableProps, TextProps, ViewProps} from 'react-native';

import {Item} from './SelectPicker';
import {SelectPickerItems} from './SelectPickerItems.android';

jest.useFakeTimers(); // jest.useFakeTimers('legacy') for jest >= 27
// call animation
jest.runAllTimers();

describe('SelectPickerItems only with required props', () => {
  it('renders successfully with more than one item', () => {
    const items = [
      {value: '1', label: 'test1'},
      {value: '2', label: 'test2'},
    ];
    render(<SelectPickerItems selectedValue="1" items={items} testID="SelectPickerItems" />);
    const selectPickerItems = screen.queryByTestId('SelectPickerItems');
    expect(screen).toMatchSnapshot('SelectPickerItems with more than one item.');
    expect(selectPickerItems).not.toBeNull();
  });

  it('renders successfully if item does not exit', () => {
    render(<SelectPickerItems selectedValue="1" items={[]} testID="SelectPickerItems" />);
    const selectPickerItems = screen.queryByTestId('SelectPickerItems');
    expect(screen).toMatchSnapshot('SelectPickerItems without item.');
    expect(selectPickerItems).not.toBeNull();
  });
});

describe('SelectPickerItems with all props', () => {
  it('should be applied properly', () => {
    const items = [
      {value: '1', label: 'test1', key: 'key1', color: 'red', fontFamily: 'Roboto'},
      {value: '2', label: 'test2', color: 'yellow', fontFamily: 'SFProText'},
      {value: '3', label: 'test3'},
      {value: '4', label: 'test4'},
      {value: '5', label: 'test5'},
      {value: '6', label: 'test6'},
    ];
    const onValueChange = jest.fn();
    const keyExtractor = jest.fn((item: Item<string>, index: number) => `${item.value}-${index}`);
    const layoutMeasurement = {height: 300};
    const contentSize = {height: 300};
    /**
     * 以下のPropは自動テストで検証できなかったため除外
     * - style
     * - itemStyle
     * - accessibilityLabel
     */
    render(
      <SelectPickerItems
        selectedValue="1"
        items={items}
        onValueChange={onValueChange}
        numberOfLines={5}
        keyExtractor={keyExtractor}
        testID="SelectPickerItems"
        itemStyle={{fontSize: 24, fontFamily: 'test-font-Regular', color: 'yellow'}}
        accessibilityLabel="testAccessibilityLabel"
        activeColor="#424242"
        inactiveColor="#565656"
        itemHeight={60}
        flatListTestID="FlatList"
        separatorTestID="Separator"
        itemPressableTestIDPrefix="itemPressable"
        itemTextTestIDPrefix="itemText"
      />,
    );
    expect(screen).toMatchSnapshot('SelectPickerItems with all props.');

    // assert selectPickerItem
    const item1Pressable = screen.getByTestId('itemPressable-1');
    const item1PressableProps = item1Pressable.props as PressableProps;
    expect(item1PressableProps.accessibilityLabel).toBe('testAccessibilityLabel');
    expect(item1PressableProps.style).toEqual({height: 60, alignItems: 'center', justifyContent: 'center'});

    const item1Text = screen.getByTestId('itemText-0');
    const item1TextProps = item1Text.props as TextProps;
    expect(item1TextProps.style).toEqual({fontSize: 24, fontFamily: 'Roboto', color: 'red'});

    const item3Text = screen.getByTestId('itemText-2');
    const item3TextProps = item3Text.props as TextProps;
    expect(item3TextProps.style).toEqual({fontSize: 24, fontFamily: 'test-font-Regular', color: 'yellow'});

    // assert flatList
    const flatList = screen.getByTestId('FlatList');
    const flatListProps = flatList.props as FlatListProps<Item<string>>;
    expect(flatListProps.initialScrollIndex).toBe(0);
    // itemsは、Itemの各Propを検証できなかったため、件数のみ検証する
    expect(flatListProps.data?.length).toBe(6);
    expect(flatListProps.data).toEqual([
      {value: '1', label: 'test1', key: 'key1', color: 'red', fontFamily: 'Roboto'},
      {value: '2', label: 'test2', color: 'yellow', fontFamily: 'SFProText'},
      {value: '3', label: 'test3'},
      {value: '4', label: 'test4'},
      {value: '5', label: 'test5'},
      {value: '6', label: 'test6'},
    ]);
    fireEvent(flatList, 'onMomentumScrollBegin');
    fireEvent(flatList, 'onMomentumScrollEnd', {
      nativeEvent: {contentOffset: {y: 180}, layoutMeasurement, contentSize},
    });
    expect(onValueChange).toHaveBeenCalled();
    expect(flatListProps.style).toEqual({height: 300, width: '100%'});
    expect(flatListProps.contentContainerStyle).toEqual({paddingVertical: 120});
    expect(flatListProps.snapToInterval).toBe(60);
    expect(keyExtractor).toHaveBeenNthCalledWith(
      1,
      {value: '1', label: 'test1', key: 'key1', color: 'red', fontFamily: 'Roboto'},
      0,
    );
    expect(keyExtractor).toHaveBeenNthCalledWith(
      2,
      {value: '2', label: 'test2', color: 'yellow', fontFamily: 'SFProText'},
      1,
    );

    // assert separator
    const separator = screen.getByTestId('Separator');
    const separatorProps = separator.props as ViewProps;
    expect(separatorProps.style).toEqual({
      position: 'absolute',
      borderRadius: 8,
      backgroundColor: 'rgba(0,0,0,0.06)',
      width: '95%',
      borderColor: 'grey',
      height: 60,
    });
  });
});
