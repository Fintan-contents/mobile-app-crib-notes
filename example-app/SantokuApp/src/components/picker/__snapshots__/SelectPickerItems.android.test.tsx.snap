// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectPickerItems only with required props renders successfully if item does not exit: SelectPickerItems without item. 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  testID="SelectPickerItems"
>
  <RCTScrollView
    animatedStyle={
      Object {
        "value": Object {},
      }
    }
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      Object {
        "paddingVertical": 88,
      }
    }
    data={Array []}
    decelerationRate={0.98}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    initialScrollIndex={-1}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    snapToInterval={44}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": 220,
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 110,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={1}
      style={
        Object {
          "height": 110,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 110,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={1}
      style={
        Object {
          "height": 110,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.06)",
        "borderColor": "grey",
        "borderRadius": 8,
        "height": 44,
        "position": "absolute",
        "width": "95%",
      }
    }
  />
</View>
`;

exports[`SelectPickerItems only with required props renders successfully with more than one item: SelectPickerItems with more than one item. 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  testID="SelectPickerItems"
>
  <RCTScrollView
    animatedStyle={
      Object {
        "value": Object {},
      }
    }
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      Object {
        "paddingVertical": 88,
      }
    }
    data={
      Array [
        Object {
          "label": "test1",
          "value": "1",
        },
        Object {
          "label": "test2",
          "value": "2",
        },
      ]
    }
    decelerationRate={0.98}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    initialScrollIndex={0}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    snapToInterval={44}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": 220,
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 44,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
              }
            }
            testID="-0"
          >
            test1
          </Text>
        </View>
      </View>
      <View
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 44,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
              }
            }
            testID="-1"
          >
            test2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 110,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={1}
      style={
        Object {
          "height": 110,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 110,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={1}
      style={
        Object {
          "height": 110,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.06)",
        "borderColor": "grey",
        "borderRadius": 8,
        "height": 44,
        "position": "absolute",
        "width": "95%",
      }
    }
  />
</View>
`;

exports[`SelectPickerItems with all props should be applied properly: SelectPickerItems with all props. 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  testID="SelectPickerItems"
>
  <RCTScrollView
    animatedStyle={
      Object {
        "value": Object {},
      }
    }
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      Object {
        "paddingVertical": 120,
      }
    }
    data={
      Array [
        Object {
          "color": "red",
          "fontFamily": "Roboto",
          "key": "key1",
          "label": "test1",
          "value": "1",
        },
        Object {
          "color": "yellow",
          "fontFamily": "SFProText",
          "label": "test2",
          "value": "2",
        },
        Object {
          "label": "test3",
          "value": "3",
        },
        Object {
          "label": "test4",
          "value": "4",
        },
        Object {
          "label": "test5",
          "value": "5",
        },
        Object {
          "label": "test6",
          "value": "6",
        },
      ]
    }
    decelerationRate={0.98}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    initialScrollIndex={0}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    snapToInterval={60}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": 300,
        "width": "100%",
      }
    }
    testID="FlatList"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={null}
      >
        <View
          accessibilityLabel="testAccessibilityLabel"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "justifyContent": "center",
            }
          }
          testID="itemPressable-0"
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontFamily": "Roboto",
                "fontSize": 24,
              }
            }
            testID="itemText-0"
          >
            test1
          </Text>
        </View>
      </View>
      <View
        style={null}
      >
        <View
          accessibilityLabel="testAccessibilityLabel"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "justifyContent": "center",
            }
          }
          testID="itemPressable-1"
        >
          <Text
            style={
              Object {
                "color": "yellow",
                "fontFamily": "SFProText",
                "fontSize": 24,
              }
            }
            testID="itemText-1"
          >
            test2
          </Text>
        </View>
      </View>
      <View
        style={null}
      >
        <View
          accessibilityLabel="testAccessibilityLabel"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "justifyContent": "center",
            }
          }
          testID="itemPressable-2"
        >
          <Text
            style={
              Object {
                "color": "yellow",
                "fontFamily": "test-font-Regular",
                "fontSize": 24,
              }
            }
            testID="itemText-2"
          >
            test3
          </Text>
        </View>
      </View>
      <View
        style={null}
      >
        <View
          accessibilityLabel="testAccessibilityLabel"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "justifyContent": "center",
            }
          }
          testID="itemPressable-3"
        >
          <Text
            style={
              Object {
                "color": "yellow",
                "fontFamily": "test-font-Regular",
                "fontSize": 24,
              }
            }
            testID="itemText-3"
          >
            test4
          </Text>
        </View>
      </View>
      <View
        style={null}
      >
        <View
          accessibilityLabel="testAccessibilityLabel"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "justifyContent": "center",
            }
          }
          testID="itemPressable-4"
        >
          <Text
            style={
              Object {
                "color": "yellow",
                "fontFamily": "test-font-Regular",
                "fontSize": 24,
              }
            }
            testID="itemText-4"
          >
            test5
          </Text>
        </View>
      </View>
      <View
        style={null}
      >
        <View
          accessibilityLabel="testAccessibilityLabel"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": 60,
              "justifyContent": "center",
            }
          }
          testID="itemPressable-5"
        >
          <Text
            style={
              Object {
                "color": "yellow",
                "fontFamily": "test-font-Regular",
                "fontSize": 24,
              }
            }
            testID="itemText-5"
          >
            test6
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 150,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={1}
      style={
        Object {
          "height": 150,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 150,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={1}
      style={
        Object {
          "height": 150,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.06)",
        "borderColor": "grey",
        "borderRadius": 8,
        "height": 60,
        "position": "absolute",
        "width": "95%",
      }
    }
    testID="Separator"
  />
</View>
`;
